python-checks/src/main/java/org/sonar/python/checks/NumpyBusDayHaveValidMaskCheck.java

```diff
@@ -0,0 +1,131 @@
+    "numpy.is_busday", 1,
+    "numpy.busdaycalendar", 0);
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, NumpyBusDayHaveValidMaskCheck::checkCallExpr);
+  }
+
+  private static void checkCallExpr(SubscriptionContext context) {
+    CallExpression callExpression = (CallExpression) context.syntaxNode();
+    Symbol calleeSymbol = callExpression.calleeSymbol();
+    if (calleeSymbol == null) {
+      return;
+    }
+    String fullyQualifiedName = calleeSymbol.fullyQualifiedName();
+    if (fullyQualifiedName == null) {
+      return;
+    }
+    Integer parameterPosition = FUNCTIONS_PARAMETER_POSITION.get(fullyQualifiedName);
+    if (parameterPosition == null) {
+      return;
+    }
+
+    RegularArgument weekmaskArgument = TreeUtils.nthArgumentOrKeyword(parameterPosition, "weekmask", callExpression.arguments());
+    if (weekmaskArgument == null) {
+      return;
+    }
+
+    if (weekmaskArgument.expression().is(Tree.Kind.STRING_LITERAL)) {
+      checkString(context, ((StringLiteral) weekmaskArgument.expression()).trimmedQuotesValue(), weekmaskArgument.expression());
+    } else if (weekmaskArgument.expression().is(Tree.Kind.LIST_LITERAL)) {
+      checkList(context, ((ListLiteral) weekmaskArgument.expression()));
+    } else if (weekmaskArgument.expression().is(Tree.Kind.NAME)) {
+      Name name = (Name) weekmaskArgument.expression();
+      Expressions.singleAssignedNonNameValue(name).ifPresent(assignedExpression -> {
+        if (assignedExpression.is(Tree.Kind.STRING_LITERAL)) {
+          checkString(context, ((StringLiteral) assignedExpression).trimmedQuotesValue(), weekmaskArgument.expression(), assignedExpression);
+        } else if (assignedExpression.is(Tree.Kind.LIST_LITERAL)) {
+          checkList(context, ((ListLiteral) assignedExpression), weekmaskArgument.expression(), assignedExpression);
+        }
+      });
+    }
+  }
+
+  private static void checkString(SubscriptionContext context, String string, Tree primaryLocation) {
+    checkString(context, string, primaryLocation, null);
+  }
+
+  private static void checkString(SubscriptionContext context, String string, Tree primaryLocation, @Nullable Tree secondaryLocation) {
+    if (PATTERN_STRING1.matcher(string).matches() || PATTERN_STRING2.matcher(string).matches()) {
+      return;
+    }
+    createIssue(context, MESSAGE_STRING, primaryLocation, secondaryLocation);
+  }
+
+  private static void checkList(SubscriptionContext context, ListLiteral listLiteral) {
+    checkList(context, listLiteral, listLiteral, null);
+  }
+
+  private static void checkList(SubscriptionContext context, ListLiteral listLiteral, Tree primaryLocation, @Nullable Tree secondaryLocation) {
+    ExpressionList listElements = listLiteral.elements();
+    List<Expression> expressionList = listElements.expressions();
+    if (expressionList.stream()
+      .filter(e -> e.is(Tree.Kind.NUMERIC_LITERAL))
```

Comment:
Here is FP if one or more item is not NUMERIC_LITERAL, you need to validate that `all` array items are numeric literals with values 0 or 1:
```java
private static final Set<String> VALID_ARRAY_WEEKMASK_VALUES = Set.of("0", "1");

  private static void checkList(SubscriptionContext context, ListLiteral listLiteral, Tree primaryLocation, @Nullable Tree secondaryLocation) {
    ExpressionList listElements = listLiteral.elements();
    List<Expression> expressionList = listElements.expressions();
    if (expressionList.size() == 7 && expressionList.stream()
      .allMatch(e -> TreeUtils.toOptionalInstanceOf(NumericLiteral.class, e)
        .map(NumericLiteral::valueAsString)
        .filter(VALID_ARRAY_WEEKMASK_VALUES::contains)
        .isPresent())) {
      return;
    }
    createIssue(context, MESSAGE_ARRAY, primaryLocation, secondaryLocation);
  }
```